crates/rpc-types/src/with_other.rs

```diff
@@ -0,0 +1,46 @@
+use crate::other::OtherFields;
+use serde::{Deserialize, Serialize};
+use std::ops::{Deref, DerefMut};
+
+/// Wrapper allowing to catch all fields missing on the inner struct while
+/// deserialize.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WithOtherFields<T> {
+    #[serde(flatten)]
+    inner: T,
```

Comment:
This needs to pub or we need another away for deconstructing all fields, currently there's no easy way to do this.

so either pub or like a split() -> (T, Other) function